### PR TITLE
Psuedoinverse function in Nx 

### PR DESCRIPTION
**Pull request by anish-lakkapragada**
`defn` Psuedoinverse function in Nx just as `np.linalg.svd` in numpy. Uses the SVD to compute the psuedoinverse.

Works only for wide matrices with backends that have SVD implemented for wide matrices

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+nx/test.ex
```
Comment (seanmor5) [inline .gitignore:1]: I think you have this leftover from debugging!

Comment (anish-lakkapragada) [reply .gitignore:1]: Sorry, I just realized that. Newbie PR here. I will create a new PR.

File: nx/lib/nx/lin_alg.ex [hunk 2]
```diff
@@ -1142,6 +1142,49 @@ defmodule Nx.LinAlg do
...
+
+
+  defn pinv(tensor, opts \\ []) do
+
+    opts = keyword!(opts, eps: @default_eps)
+
+    {u, s, vt} = Nx.LinAlg.svd(tensor)
+
+    s_shape = {elem(Nx.shape(s), 0), elem(Nx.shape(u, 1)}
+    adjusted_s = Nx.map(s, fn x -> if Nx.abs(x) < opts[:eps] do 0 else Nx.divide(1, x) end end)
```
Comment (seanmor5) [inline nx/lib/nx/lin_alg.ex:1182]: `Nx.select(Nx.abs(x) < opts[:eps], 0, Nx.divide(1, x))`

File: .gitignore (hunk 1 above)
Comment (seanmor5) [reply .gitignore:1]: There is no need, you can just update your fork and it will reflect here :)

**Pull request closed without merging**